docs/HTTP-batchsource.md

```diff
@@ -0,0 +1,206 @@
+
+### General
+
+**URL:** Url to fetch to the first page.
+The url must start with a protocol (e.g. http://).
+
+Supports {pagination.index} placeholder, neccessary for pagination
+type "Increment an index". For more information read pagination section.
+
+**HTTP Method:** Request HTTP method.
+
+**Headers:** Request HTTP headers for the.
+
+**Request body:** A body of the request.
+
+### Format
+
+**Format:** Format of pages. Used to determine how to convert pages into output records. Possible values are:<br>
+JSON - retrieves all records from given location (json path)
+and transforms them into records according to the mapping.<br>
+XML - retrieves all records from given location (XPath)
+and transforms them into records according to the mapping.<br>
+TSV/CSV - transforms TSV/CSV pages into records. Page columns are mapped to record fields in the order they are
+listed in schema.<br>
+Text - transforms a single line of text into a single record with a string field "body" containing the result.<br>
+BLOB - transforms a single page into a single record with a byte array field "body" containing the result.
+
+**JSON/XML Result Path:** Path to the results. When XML is used this is an XPath, when json is used this is a JSON path.
+JSON path examples: "/bookstore/books", "/feed/entries/items". JSON path should either point to an array in a json
+structure or to a field within an array element.
+
+**JSON/XML Fields Mapping:** Mapping of fields in a record to fields in retrieved element. The left column contains the
+name of schema field. The right column contains path to it within a relative to an element. It can be either XPath or 
+JSON path.
+
+Example:
+
+| Field Name      | Field Path                                |
+| --------------- |:-----------------------------------------:|
+| name            | /key                                      |
+| type            | /fields/issuetype/name                    |
+| description     | /fields/description                       |
+| projectCategory | /fields/project/projectCategory/name      |
+| isSubtask       | /fields/issuetype/subtask                 |
+| fixVersions     | /fields/fixVersions                       |
+
+**CSV Skip First Row:** If true, skip first row. This should be set in case first row of csv is a header row.
+
+### Basic Authentication
+
+**Username:** Username for basic authentication.
+
+**Password:** Password for basic authentication.
+
+### HTTP Proxy
+
+**Proxy URL:** Proxy URL. Must contain a protocol, address and port.
+
+**Username:** Proxy username.
+
+**Password:** Proxy password.
+
+### Error Handling
+
+**HTTP Errors Handling:** A mapping which defines error handling strategy used, when pages return certain HTTP status
+codes. The left column contains a regular expression for http status code. The right column contains an action which
+is done in case of match. If http status code matches multiple regular expressions, the first specified in mapping
```

Comment:
If http status code ... -> If the HTTP status code matches multiple regular expressions, the first one is used.

Reply:
done